src/wallet/rpcwallet.cpp

```diff
@@ -2737,16 +2737,23 @@ CScriptID calculate_contract(const CScript& federationRedeemScript, const CBitco
     assert(opcodeTmp == 0x14 && destHash.size() == 20);
     memcpy(fullcontract + 4 + 16, &destHash[0], destHash.size());
 
+    // Catch blank or OP_TRUE fedpegscripts which are static and unsafe to peg into
+    bool nokeys = true;
+
     CScript scriptDestination(federationRedeemScript);
     {
         CScript::iterator sdpc = scriptDestination.begin();
         vector<unsigned char> vch;
         while (scriptDestination.GetOp(sdpc, opcodeTmp, vch))
         {
-            assert((vch.size() == 33 && opcodeTmp < OP_PUSHDATA4) ||
-                   (opcodeTmp <= OP_16 && opcodeTmp >= OP_1) || opcodeTmp == OP_CHECKMULTISIG);
+            if(!((vch.size() == 33 && opcodeTmp < OP_PUSHDATA4) ||
```

Comment:
This predicate makes my eyes hurt. From what I can tell it is suppose to make sure we only have valid multisig script operations. Can't we factor this out into some sort of function and call it `IsMultiSigScript` or something like that?